Christmas cleanup and modularisations

guillep opened the pull request:
- moving Dependent out of object
- removing dead classes
- moving date related classes to the chronology package

guillep committed:
Remove dead #BytecodeMappedInlinePrimitiveConfiguration and hooks

guillep committed:
Remove DependentsFields from Object

guillep committed:
Extracting chronology into its own package

guillep committed:
Moving Chronology classes to chronology package

Ducasse commented:
Did not bootstrap.

Ducasse commented:
```
Looking for module  ... loaded...computing accessor depths17:00:38		[+] Creating image on which bootstrapping will take place

warning, variable nsMethodCache doesn't exist or has already been removed
...done17:00:41		Initializing bootstrap environment.
17:00:41		Creating initial Objects needed by the VM
17:00:41		Flushing new space.
17:00:41				Begin: Initializing initial objects
17:00:41				Initializing objects.
17:00:41				initializing hash table sizes
17:00:41				preparing class builder
17:00:42				class loader now creates class pools
17:00:42				Initializing StartUp list
17:00:42				Initializing Character Table
17:00:42				Initializing String Ascii and CaseInsensitive Table
NewUndeclaredWarning: ProtoObject>>noMethod (DateAndTime is Undeclared)
Instance of EPASTInterpreter did not understand #visitLiteralVariableNode:
EPASTInterpreter(Object)>>doesNotUnderstand: #visitLiteralVariableNode:
UndeclaredVariable(LiteralVariable)>>acceptVisitor:node:
RBVariableNode>>acceptVisitor:
EPASTInterpreter(ASTInterpreter)>>interpret:
EPASTInterpreter>>visitMessageNode:
RBMessageNode>>acceptVisitor:
EPASTInterpreter(ASTInterpreter)>>interpret:
EPASTInterpreter>>evaluateAST:
EPASTEvaluator>>evaluateAST:
EPASTEvaluator>>evaluateCode:withTemps:
EPASTEvaluator>>evaluateCode:
17:00:43				Initializing SmallInteger constants
17:00:43				Initializing Date and Time constants
[
```

guillep committed:
Remove date initialization from bootstrap

guillep committed:
Remove dependency from system-announcements to chronology

guillep committed:
Do not initialize version at startup

guillep committed:
Move PerformMessageCommandLineHandler closer to the bootstrap as a minimal dependency and versatile command line handler

guillep committed:
Add version creation method to use at bootstrap time

guillep committed:
Push system version creation later during bootstrap

guillep committed:
Fix bash command to use {}

guillep committed:
Small hermes cleanup

guillep committed:
Do not initialize Object, Date nor Chronology constants.
Object does not require an initialization anymore.
Date and chronology constants will be loaded and initialized later

guillep committed:
Do not initialize collection: it does not define initialize

guillep committed:
Remove usage of date/time in session

guillep committed:
No need to ask for Pharo12, we are in the Pharo12 branch

guillep committed:
Use P12 package API

guillep committed:
Use P12 method creation API in P12 branch

guillep committed:
initialize chronology classes

guillep committed:
Do not initialize FileHandle that does not redefine #initialize.
Print result in perform command line handler

guillep committed:
Guarantee versions are integers

guillep committed:
Cut a dependency on Display

guillep committed:
Do not initialize classes without initialize method

guillep committed:
Do not initialize Text

guillep committed:
Make calypso reference Number tag

guillep committed:
Make calypso tests depends on calypso classes

guillep committed:
Fix ClassTest

guillep committed:
Fix tests:
 - categorizations
 - pass on tests depending on existing class variables

guillep committed:
Move deep copier closer to Morphic through Model

guillep committed:
Remove more class side super initialize

Ducasse commented on src/System-SessionManager/WorkingSession.class.st:
```diff
@@ -64,7 +58,6 @@ WorkingSession >> executeDeferredStartupActions: resuming [
 WorkingSession >> initialize [
 	super initialize.
 	deferredStartupActions := OrderedCollection new.
-	creationTime := DateAndTime now.
```
I wondered if this was used. 
In the redefinition later will we reintroduce it?

Ducasse replied:
Probably for sorting them.

guillep replied:
Nope, not even...

Ducasse commented:
The class side initialize are surprising to me. Did you check the complete system?

Ducasse commented:
I was thinking about the `Delay forDuration: ` and I will open an issue. Because we have totally stupid code

```
indicateModalChild
	"Flash the button border."

	|fs c w d|
	fs := self fillStyle.
	c := self color alphaMixed: 0.5 with: Color black.
	w := self world.
	d := 0.
	self assureExtension.
	2 timesRepeat: [
		(Delay forDuration: d milliSeconds) wait.
		d := 200.
		extension fillStyle: c.
		self privateColor: c.
		self invalidRect: self bounds.
		w ifNotNil: [w displayWorldSafely].
		(Delay forDuration: d milliSeconds) wait.
		self fillStyle: fs.
		w ifNotNil: [w displayWorldSafely].
		self invalidRect: self bounds]
```

guillep commented:
> The class side initialize are surprising to me. Did you check the complete system?

Nope, but I'm positive I caught all of them:

It used to work because `Object class >> initialize` was defined (and resetting the dependents array!).
Since we removed that method, now lookup goes up and clears the class, so it seems that I caught all the extra initializes because the build works :)

guillep committed:
Fix remaining tests

guillep committed:
Initialize date and time

guillep committed:
Add System-Model in kernel test run

guillep committed:
Add system-model to the list of sunit dependencies

guillep committed:
Merge remote-tracking branch 'pharo-project/Pharo12' into bottom

guillep commented:
Broken tests are unrelated

guillep commented:
Ok, this has introduced a problem when starting up that is not shown in the tests ... I'll revert and rework.
